models/rate_neuron_opn_impl.h

```diff
@@ -257,13 +261,25 @@ nest::rate_neuron_opn< TGainfunction >::update_( Time const& origin,
     {
       if ( P_.linear_summation_ )
       {
-        S_.rate_ += V_.P2_ * gain_( B_.delayed_rates_.get_value_wfr_update(
-                                      lag ) + B_.instant_rates_[ lag ] );
+        S_.rate_ += V_.P2_
+          * nonlinearities_.mult_coupling_ex( new_rates[ lag ] )
+          * nonlinearities_.input( B_.delayed_rates_ex_.get_value_wfr_update(
+                                     lag ) + B_.instant_rates_ex_[ lag ] );
+        S_.rate_ += V_.P2_
+          * nonlinearities_.mult_coupling_in( new_rates[ lag ] )
+          * nonlinearities_.input( B_.delayed_rates_in_.get_value_wfr_update(
+                                     lag ) + B_.instant_rates_in_[ lag ] );
       }
```

Comment:
Nice fix. This makes the code consistent with the old behavior. The only issue I still have is the following:
If I thought of the simplest nonlinear rate model with multiplicative coupling, I would write down a term like

f(rate)*g(input)

where f = mult_coupling_ex = mult_coupling_in and input = exc_input + inh_input.
This model is not captured by your code which is OK. But I think that the documentation of the models should clarify this. In particular it should become clear to the user that in case of multiplicative coupling and mult_coupling_ex = mult_coupling_in linear summation still does not sum up excitatory and inhibitory inputs linearly.
This would also maybe suggest to rename the boolean linear_summation to make it less confusing.
What do you think?
Sorry for being so picky...

Reply:
@ddahmen You are right that this simplest multiplicative coupling can only be achieved with a trick (connect all incoming inhibitory connections with a `lin_rate_parrot` neuron with the desired weight and then connect this helper neuron with the target neuron with weight 1.0 - then the mult_coupling_ex is the joint factor for the multiplicative coupling).
We can include this in the documentation somewhere - however at the present time I don't know where to put it: The only model that has multiplicative coupling implemented is the lin_rate neuron, for which g(input_exc) + g(input_inh) = g(input_exc + input_inh) holds anyway.

I would also suggest to keep the name `linear_summation`, for consistency to 2.14 and since - given two different multiplicative coupling factors for exh and inh coupling - it should be clear that the linear summation can only happen within the two subsets (or is there some way that I do not think of at the moment?)

Reply:
You are right that the naming better be consistent with NEST 2.14. I also agree that for lin_rate the above mentioned problem is irrelevant. The only problem could be that in the future somebody will create another neuron model from the templates and does not keep the above issue in mind. One could help future developers by adding the following sentence to the documentation of rate_neuron_opn_impl and rate_neuron_ipn_impl:

"The boolean parameter linear_summation
 determines whether the input function is applied to the summed up incoming
 connections (= True, default value) or to each input individually (= False). In case of multiplicative coupling the nonlinearity is applied separately to the summed excitatory and inhibitory inputs if linear_summation=True."

Reply:
I added the suggested additional comment